comment-updater-core/src/main/kotlin/org/jetbrains/research/commentupdater/models/metrics/SimilarityModel.kt

```diff
@@ -0,0 +1,110 @@
+package org.jetbrains.research.commentupdater.models.metrics
+
+import ai.onnxruntime.OnnxTensor
+import ai.onnxruntime.OrtEnvironment
+import ai.onnxruntime.OrtSession
+import com.beust.klaxon.Klaxon
+import org.jetbrains.research.commentupdater.models.ONNXTensorUtils
+import org.jetbrains.research.commentupdater.models.config.EmbeddingConfig
+import org.jetbrains.research.commentupdater.models.config.ModelFilesConfig
+import org.ojalgo.array.Primitive64Array
+import org.ojalgo.function.constant.PrimitiveMath
+import org.ojalgo.matrix.Primitive64Matrix
+import org.ojalgo.matrix.store.Primitive64Store
+import java.io.File
+
+class SimilarityModel {
+
+    val embeddingConfig: EmbeddingConfig
+    val env: OrtEnvironment
+    val codeEmbeddingsSession: OrtSession
+    val commentEmbeddingSession: OrtSession
+    val EMBEDDING_SIZE = 64
+
+    init {
+        embeddingConfig = Klaxon().parse<EmbeddingConfig>(File(ModelFilesConfig.EMBEDDING_FILE))
+            ?: throw Exception("can't load embeddings")
+
+
+        env = OrtEnvironment.getEnvironment()
+        codeEmbeddingsSession = env.createSession(ModelFilesConfig.CODE_EMBEDDING_ONNX_FILE, OrtSession.SessionOptions())
+        commentEmbeddingSession = env.createSession(ModelFilesConfig.COMMENT_EMBEDDING_ONNX_FILE, OrtSession.SessionOptions())
+    }
+
+    fun compute(tokens1: List<String>, tokens2: List<String>, useCodeVocab: Boolean): Double {
```

Comment:
what does it compute? please, add a comment for this method, it does a lot of things and it will be much easier to understand it if we add a comment.